raster-foundry/raster-foundry
# Add SceneToLayerDao; Update ProjectDao and Project Endpoints

## Overview

This PR:
 * adds:
     - a migration to populate `scenes_to_layers` table with projects' default layer ids associated with their scenes
     - `SceneToLayerDao` and its property tests that mirror the previous `SceneToProjectDao`
     - `SceneToLayer` data model
 * updates:
     - `ProjectDao` so that adding/replacing/deleting scenes from projects will insert/update/delete from both `scenes_to_layers` and `scenes_to_projects` tables (the latter is still preserved since we can't drop the table and the dao yet)
     - project endpoints so that it uses both `SceneToLayerDao` and `SceneToProjectDao` (same reason as the above)

**Updated on 01/23/2019:**
It also makes scene ordering consistent when mosaic-ing project scenes together and getting the project scene list


### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

## Testing Instructions

 * Run migration
 * CI should pass
 * Create a new project, check that if a default layer is created in this project's record, and a new record in `project_layers` is created to associate this project with the default layer
 * Add/update/delete scenes to/in/from this project -- make sure it still works, and observe that these operations affect both `scenes_to_layers` and `scenes_to_projects` tables.
 * The previous step should operate scenes all on the project's default layer

Closes https://github.com/raster-foundry/raster-foundry/issues/4484
Closes https://github.com/raster-foundry/raster-foundry/issues/4477
